hash/hashes.go

```diff
@@ -57,26 +62,16 @@ var digestSize = []uint8{
 	MIMC_BW6_633:   80,
 }
 
-// New initializes the hash function.
+// New initializes the hash function. This is a convenience function which does
+// not allow setting hash-specific options.
 func (m Hash) New() hash.Hash {
-	switch m {
-	case MIMC_BN254:
-		return bn254.NewMiMC()
-	case MIMC_BLS12_381:
-		return bls381.NewMiMC()
-	case MIMC_BLS12_377:
-		return bls377.NewMiMC()
-	case MIMC_BW6_761:
-		return bw761.NewMiMC()
-	case MIMC_BLS24_315:
-		return bls315.NewMiMC()
-	case MIMC_BLS24_317:
-		return bls317.NewMiMC()
-	case MIMC_BW6_633:
-		return bw633.NewMiMC()
-	default:
-		panic("Unknown mimc ID")
+	if m < maxHash {
+		f := hashes[m]
+		if f != nil {
+			return f()
+		}
 	}
+	panic("requested hash function #" + m.String() + " not registered")
```

Comment:
since it's going to break existing users code bases, can we make their life easier by augmenting this message with a suggestion containing the full "consider adding "import _ ..../strings.ToLower(m.String())/ ... " or whatever ?

Reply:
Fixed